fedimint-server/src/atomic_broadcast/broadcast.rs

```diff
@@ -44,79 +50,95 @@ impl AtomicBroadcast {
-
-                    ordered_item_sender
-                        .send(Some((ordered_item, decision_sender)))
-                        .await
-                        .ok();
-
-                    decision_receivers.push(decision_receiver);
-                }
-
-                // signal that the session is complete
-                ordered_item_sender.send(None).await.ok();
-
-                for decision_receiver in decision_receivers {
-                    // The items in a threshold signed block have to be accepted
-                    if let Ok(decision) = decision_receiver.await {
-                        assert_eq!(decision, Decision::Accept)
-                    }
-                }
-            });
-        } else {
-            tracing::info!(target: LOG_CONSENSUS,"Run session with index {}", index);
-
-            let (backup_loader, backup_saver) = db::open_session(self.db.clone(), index).await;
-
-            let keychain = self.keychain.clone();
-            let db = self.db.clone();
-            let mempool_item_receiver = self.mempool_item_receiver.clone();
-            let incoming_message_receiver = self.incoming_message_receiver.clone();
-            let outgoing_message_sender = self.outgoing_message_sender.clone();
-
-            spawn("atomic run session", async move {
-                let session_result = session::run(
-                    index,
-                    keychain,
-                    backup_loader,
-                    backup_saver,
-                    mempool_item_receiver,
-                    incoming_message_receiver,
-                    outgoing_message_sender,
-                    ordered_item_sender.clone(),
-                    federation_api,
-                );
-
-                if let Ok(signed_block) = session_result.await {
-                    db::complete_session(&db, index, signed_block.clone()).await;
-
-                    tracing::info!(target: LOG_CONSENSUS,"Completed session with index {}", index);
-
-                    // signal that the session is complete. It is critical that we do so only after
-                    // we call db::complete_session.
-                    // Otherwise the broadcast may miss a signed block in its history if Fedimint
-                    // Consensus signal first and the system
-                    // crashes before the signed block is stable on disk.
-                    ordered_item_sender.send(None).await.ok();
-                } else {
-                    tracing::warn!(target: LOG_CONSENSUS,"Session with index {} has been interrupted", index);
-                }
-            });
-        }
-
-        ordered_item_receiver
+    ) -> anyhow::Result<()> {
+        // if all nodes are correct the session will take 45 to 60 seconds. The
+        // more nodes go offline the longer the session will take to complete.
+        const EXPECTED_ROUNDS_PER_SESSION: usize = 45 * 4;
```

Comment:
**suggestion (non-blocking):** Units unclear. Would be best to use `Duration`, but not possible / doesn't seem like a good idea at least `*_SECS` etc. would help.